plugins/linux-lockdown/fu-linux-lockdown-plugin.c

```diff
@@ -85,18 +85,75 @@ fu_linux_lockdown_plugin_startup(FuPlugin *plugin, FuProgress *progress, GError
 	return TRUE;
 }
 
+static gboolean
+fu_linux_lockdown_plugin_ensure_security_attr_flags(FwupdSecurityAttr *attr, GError **error)
+{
+	const gchar *value;
+	g_autoptr(GHashTable) cmdline = NULL;
+	g_autoptr(GHashTable) config = NULL;
+
+	/* get current args */
+	cmdline = fu_kernel_get_cmdline(error);
+	if (cmdline == NULL)
+		return FALSE;
+
+	/* can we modify the args */
+	if (!fu_kernel_check_cmdline_mutable(error))
+		return FALSE;
+
+	/* get build flags */
+	config = fu_kernel_get_config(error);
+	if (config == NULL)
+		return FALSE;
+	if (!g_hash_table_contains(config, "CONFIG_LOCK_DOWN_KERNEL_FORCE_NONE")) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_NOT_SUPPORTED,
+				    "config does not have CONFIG_LOCK_DOWN_KERNEL_FORCE_NONE");
+		return FALSE;
+	}
+
+	/* we cannot change this */
+	if (g_hash_table_contains(config, "CONFIG_LOCK_DOWN_IN_EFI_SECURE_BOOT") &&
+	    fu_efivar_secure_boot_enabled(NULL)) {
+		g_set_error_literal(
+		    error,
+		    G_IO_ERROR,
+		    G_IO_ERROR_INVALID_ARGUMENT,
+		    "kernel lockdown cannot be changed when secure boot is enabled");
+		return FALSE;
+	}
+
+	/* set the current and target values */
+	value = g_hash_table_lookup(cmdline, "lockdown");
+	fwupd_security_attr_set_kernel_current_value(attr, value);
+	if (g_strcmp0(value, "integrity") != 0) {
+		fwupd_security_attr_set_kernel_target_value(attr, "lockdown=integrity");
```

Comment:
you're inconsistent here use `integrity` and later use `confidentiality`.  Shouldn't they be consistent?

Furthermore it's a corner case but what if they start with nothing forced in kernel and they have `lockdown=integrity` on kernel command line and it gets fixed to `lockdown=confidentiality` but then unfixed to `lockdown=none`?  So I wonder if the old value needs to be stored to the history database.

Reply:
Aha, so we don't want to use confidentiality -- according to https://mjg59.dreamwidth.org/55105.html "_Don't use confidentiality mode in general purpose distributions_" -- I'll switch this to integrity everywhere.